File: deployment/initcontainer/tengu_init.py
import os
import signal


def main():
    for key in os.environ.keys():
        print("{} = {}".format(key, os.environ[key]))

    required_vars = os.environ['TENGU_REQUIRED_VARS']

    for var in required_vars.split(','):
        if var not in os.environ:
            print("Var not found ({}) -> BLOCKING".format(var))
            signal.pause()
        
if __name__ == '__main__':
    main()

File: deployment/initcontainer/tengu-init.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"strings"
	"syscall"
)

func main() {
	requiredVars := os.Getenv("TENGU_REQUIRED_VARS")
	for _, requiredVar := range strings.Split(requiredVars, ",") {
		if _, ok := os.LookupEnv(requiredVar); !ok {
			fmt.Printf("Var not found (%v) -> BLOCKING\n", requiredVar)

			// Sleeping forever
			// https://stackoverflow.com/a/36419288/1588555
			exitSignal := make(chan os.Signal)
			signal.Notify(exitSignal, syscall.SIGINT, syscall.SIGTERM)
			<-exitSignal
		}
	}
	fmt.Println("All variables found; shutting down..")
}
